### PR TITLE
Address analyzer suggestions for csharp-project

### PR DESCRIPTION
**Pull request by fflaten**
## PR Summary
Now that language version is increased to latest, we can simplify some of our C# code using newer features.

This fixes most findings reported by Roslyn analyzer.

## PR Checklist

- [x] PR has meaningful title
- [x] Summary describes changes
- [x] PR is ready to be merged
  - If not, use the arrow next to `Create Pull Request` to mark it as a draft. PR can be marked `Ready for review` when it's ready.
- [ ] Tests are added/update *(if required)*
- [ ] Documentation is updated/added *(if required)*

### CODE REVIEW

**Comment by fflaten** [comment]
This was mostly default Roslyn analyzer findings. Besides blank lines, I skipped the Roslynator suggestions for now (see below). Should I add a `.editorconfig` to enforce some style patterns and fix the rest?

image

**Comment by fflaten** [comment]
Any way to identify which features that require newer runtime? Since we're using newer language than recommended for net452 (7.3) atm. 

> This default choice also ensures you don't use a language that requires types or runtime behavior not available in your target framework. Choosing a language version newer than the default can cause hard to diagnose compile-time and runtime errors.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version

We've mostly used "synthetic sugar" that should be compiled compatible yet AFAIK.

**Comment by nohwnd** [comment]
We have been using langVersion latest in all of our projects, and until recently were compiling down to net452, and we never seen this cause any trouble, so I would not worry about it.

File: src/csharp/Pester/Factory.cs [hunk 1]
```diff
@@ -42,16 +42,16 @@ public static ErrorRecord CreateShouldErrorRecord(string message, string file, s
...
-            targetObject.Add("Terminating", terminating);
-            var errorRecord = new ErrorRecord(exception, errorId, errorCategory, targetObject);
-            return errorRecord;
+            var targetObject = new Dictionary<string, object>
+            {
+                { "Message", message },
+                { "File", file },
+                { "Line", line },
+                { "LineText", lineText },
+                { "Terminating", terminating }
```
Comment (nohwnd) [inline src/csharp/Pester/Factory.cs:52]:
```suggestion
                ["Message"] = message,
                ["File"] = file,
                ["Line"] = line,
                ["LineText"] = lineText,
                ["Terminating"] = terminating,
```

I find this syntax little more obvious.

File: src/csharp/Pester/Test.cs [hunk 2]
```diff
@@ -18,11 +18,15 @@ public Test()
...
 
-            var runtime = new Hashtable();
-            runtime.Add("Phase", null);
-            runtime.Add("ExecutionStep", null);
-            FrameworkData = new Hashtable();
-            FrameworkData.Add("Runtime", runtime);
+            var runtime = new Hashtable
+            {
+                { "Phase", null },
+                { "ExecutionStep", null }
```
Comment (nohwnd) [inline src/csharp/Pester/Test.cs:24]:
```suggestion
                ["Phase"] = null,
                ["ExecutionStep"] = null,
```

not 100% sure this works for hashtables.

File: src/csharp/Pester/Test.cs [hunk 3]
```diff
@@ -18,11 +18,15 @@ public Test()
...
+            };
+            FrameworkData = new Hashtable
+            {
+                { "Runtime", runtime }
```
Comment (nohwnd) [inline src/csharp/Pester/Test.cs:28]:
```suggestion
                ["Runtime"] = runtime,
```

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
let me know if you want to apply those nitpicks.

**Comment by fflaten** [comment]
> let me know if you want to apply those nitpicks.

Don't think I've seen that before. More PowerShell'y 🙂

**Pull request merged**